core/jobs/batch_jobs/manual_voice_artist_name_job.py

```diff
@@ -118,57 +114,69 @@ def extract_added_voiceovers_between_successive_snapshots(
+            Exception. Failed to get newly added voiceovers between the given
+                snapshot models.
+        """
+        try:
+            # The voiceover mapping that is present in the new snapshot model.
+            new_voiceover_mapping: Dict[str, Dict[
+                str, state_domain.VoiceoverDict]] = {}
+
+            # The voiceover mapping that is present in the old snapshot model.
+            old_voiceover_mapping: Dict[str, Dict[
+                str, state_domain.VoiceoverDict]] = {}
+
+            for state in new_snapshot_model.content['states'].values():
+                new_voiceover_mapping.update(
+                    state['recorded_voiceovers']['voiceovers_mapping'])
+
+            for state in old_snapshot_model.content['states'].values():
+                old_voiceover_mapping.update(
+                    state['recorded_voiceovers']['voiceovers_mapping'])
+
+            # The voiceover mapping that has been added to this version of the
+            # exploration snapshot.
+            voiceovers_added_in_this_version: Dict[
+                str, Dict[str, state_domain.VoiceoverDict]] = (
+                    collections.defaultdict(dict)
+                )
 
-        for content_id, lang_code_to_voiceover_dict in (
-                new_voiceover_mapping.items()):
-            for lang_code, voiceover_dict in (
-                    lang_code_to_voiceover_dict.items()):
-                if lang_code not in old_voiceover_mapping.get(content_id, {}):
-                    voiceovers_added_in_this_version[content_id][lang_code] = (
-                        voiceover_dict)
-                else:
-                    old_voiceover_dict = old_voiceover_mapping[
-                        content_id][lang_code]
-                    new_voiceover_dict = lang_code_to_voiceover_dict[lang_code]
-
-                    if old_voiceover_dict != new_voiceover_dict:
+            for content_id, lang_code_to_voiceover_dict in (
+                    new_voiceover_mapping.items()):
+                for lang_code, voiceover_dict in (
+                        lang_code_to_voiceover_dict.items()):
+                    if lang_code not in old_voiceover_mapping.get(
+                            content_id, {}):
                         voiceovers_added_in_this_version[
                             content_id][lang_code] = voiceover_dict
+                    else:
+                        old_voiceover_dict = old_voiceover_mapping[
+                            content_id][lang_code]
+                        new_voiceover_dict = (
+                            lang_code_to_voiceover_dict[lang_code])
+
+                        if old_voiceover_dict != new_voiceover_dict:
+                            voiceovers_added_in_this_version[
+                                content_id][lang_code] = voiceover_dict
 
-        return voiceovers_added_in_this_version
+            return voiceovers_added_in_this_version
+        except Exception as e:
```

Comment:
Why do we need this, exactly? I worry that the try/except here is much too broad, is there some actual error we should be handling here?